### PR TITLE
Feature/ch14

### PR DESCRIPTION
**Pull request by monkeywithbanana**

### CODE REVIEW

File: 14/AutoResetTimerExample.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#include "AutoResetTimerExample.h"
+
+using namespace std;
+
+namespace samples
+{
+	static mutex sMutex;
+	static condition_variable mEvent;
+	static int seconds;
+	const int MAX_SECONDS = 10;
```
Comment (popekim) [inline 14/AutoResetTimerExample.cpp]: constexpr?

File: 14/AutoResetTimerExample.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
+#include <mutex>
+#include <iostream>
+#include "AutoResetTimerExample.h"
+
+using namespace std;
+
+namespace samples
+{
+	static mutex sMutex;
+	static condition_variable mEvent;
```
Comment (popekim) [inline 14/AutoResetTimerExample.cpp]: why one is s.. and another is m?

both s or both m

File: 14/AutoResetTimerExample.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		while (true)
+		{
+			{
+				scoped_lock<mutex> lock(sMutex);
+				cout << seconds << endl;
+			}
+
+			chrono::seconds delay = chrono::seconds(1);
+			auto end = chrono::high_resolution_clock::now() + delay;
+			while (chrono::high_resolution_clock::now() < end)
```
Comment (popekim) [inline 14/AutoResetTimerExample.cpp]: you can just do sleep for?

Comment (popekim) [reply 14/AutoResetTimerExample.cpp]: thought that sleep for is not really 1 sec on spot. is it intended?

File: 14/AutoResetTimerExample.cpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+			{
+				unique_lock<mutex> lock(sMutex);
+				mEvent.wait(lock, [] { return seconds >= MAX_SECONDS; });
+
+				seconds = 0;
+			}
+
+			{
+				scoped_lock<mutex> lock(sMutex);
+				cout << "Reset: " << seconds << endl;
```
Comment (popekim) [inline 14/AutoResetTimerExample.cpp:45]: move this line to after line # 47.

u don't want to unlock and lock here. (when wait() wakes up. it does auto locking that is

File: 14/MultiThreadingExample.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+		{
+			scoped_lock<mutex> lock(sMutex);
+			cout << "Executing SlowAdd" << endl;
+		}
+
+		chrono::seconds delay = chrono::seconds(1);
+		auto end = chrono::high_resolution_clock::now() + delay;
+		while (chrono::high_resolution_clock::now() < end)
+		{
+			this_thread::yield();
```
Comment (popekim) [inline 14/MultiThreadingExample.cpp]: same comment as above example

**Review by popekim — APPROVED** [review]
merge after fix

File: 14/AutoResetTimerExample.cpp (hunk 3 above)
Comment (monkeywithbanana) [reply 14/AutoResetTimerExample.cpp]: I thought sleep_for was thread blocking?

**Pull request merged**